xapian-core/cluster/cluster.cc

```diff
@@ -73,21 +80,10 @@ TermListGroup::add_document(const Document &document) {
     for (; titer != end; ++titer) {
 	unordered_map<string, doccount>::iterator i;
 	i = termfreq.find(*titer);
-	if (i == termfreq.end())
-	    termfreq[*titer] = 1;
-	else
-	    i->second += 1;
+	(i == termfreq.end()) ? termfreq[*titer] = 1 : i->second += 1;
```

Comment:
I'm not convinced this is more readable. The ternary operator is fine for an expression, but you have side-effects here; indeed the side-effects are the point. So the original `if…else` is probably clearer.